Source/Urho3D/Engine/Engine.cpp

```diff
@@ -355,7 +355,11 @@ bool Engine::Initialize(const StringVariantMap& parameters)
         if (HasParameter(EP_WINDOW_MAXIMIZE) && GetParameter(EP_WINDOW_MAXIMIZE).GetBool())
             graphics->Maximize();
 
-        graphics->SetShaderCacheDir(GetParameter(EP_SHADER_CACHE_DIR).GetString() + "shadercache/");
+        ea::string shaderCacheDir = GetParameter(EP_SHADER_CACHE_DIR).GetString();
+        if (shaderCacheDir.empty())
+            shaderCacheDir = appPreferencesDir_;
```

Comment:
Since `engineParameters_` now defines defaults, maybe `appPreferencesDir_` could be a default value for `EP_SHADER_CACHE_DIR` then?